README: Express preference for Github discussions for questions

mxinden opened the pull request:
I have been answering questions over at https://discuss.libp2p.io/ in the last couple of weeks. While I appreciate having the forum for cross-language topics, I would prefer to answer anything rust-libp2p specific in this repository directly. Objections?

Slightly related: Is anyone active in #libp2p on IRC? I am not, thus I am not sure whether we should still list it as an option.

romanb commented:
For questions the [discussions](https://github.com/libp2p/rust-libp2p/discussions) here on github seem more appropriate than creating issues.

mxinden commented:
> For questions the [discussions](https://github.com/libp2p/rust-libp2p/discussions) here on github seem more appropriate than creating issues.

I don't have any experience with GitHub's _Discussions_, but happy to give it a shot. I added it to the README.md.

In addition I removed the mention of the IRC channel. Happy to revert it in case people are active on the channel.
